Feat/saf 24

AgathaLice opened the pull request:
O timer está pronto, completamente funcional e modificado para ser implementado com o que foi me pedido além do Jira, para implementação.
Qualquer necessidade me avise para mudar, embora funcional, ainda precisa ser decidido o som de final, caso seja importado. Também, não tenho certeza se usei corretamente todas as boas práticas com globais, por exemplo.
Embora isso, já confirmei que é funcional.

AgathaLice committed:
Update timer.py

AgathaLice committed:
Update timer.py

Correção em timer, agora pode ser implementado, falta adicionar winsound e limpar as linhas de teste.

AgathaLice committed:
Testes som

Testes de som de final de jogo.

AgathaLice committed:
feat: pausar timer

Adicionei a função para pausar o timer

AgathaLice committed:
updates timer

implementado som, retirado linhas de teste e comentários desnecessários

Tybika commented on src/utils/timer.py:
```diff
@@ -1,37 +1,56 @@
-#! Esse é o timer em si, o código de como ele funciona
-def timer(tempo_Segundos): 
-    contador = tempo_Segundos # Não sabia se era uma boa ideia modificar um parâmetro diretamente
-    if contador >= 0: # O timer vai do tempo enviado até zero (quando acaba o jogo)
-        sleep(1) # Ele usa esse método do import time para aguardar um segundo
-#        print(segundo_Para_Minuto(contador)) #########!Essa linha é para teste, não é necessária o print, apenas a chamada da função, caso queira testar o timer, modifique o valor de segundos abaixo e retire o "#" do começo da linha
-        contador -= 1 # Já que se passaram um segundo, ele reduz em um o contador, pois também é uma função recursiva
-        timer(contador) # Chama a si mesma, dando o contador como parâmetro, agora que penso, ainda estou tentando lembrar por que não usei um for loop in range, tinha um motivo bom, juro
-    else:
-        return False # Quando o tempo chega a zero, o timer acaba
+        return segundo + "s" #! 00s
```
só para uniformizar os retornos da função segundo_Para_Minuto(), poderia mudar a redação para:
`def segundo_Para_Minuto(tempo): # Tem como parâmetro o tempo a ser traduzido
    minuto = floor(tempo / 60)
    segundo = tempo % 60
    minuto = minuto
    if minuto >= 1 and segundo >= 10:
        return "0" + str(minuto) + ":" + str(segundo) #! 00:00
    elif minuto >= 1 and segundo < 10:
        return "0" + str(minuto) + ":0" + str(segundo) #! 00:00
    elif minuto < 1 and segundo >= 10:
        return "00:" + str(segundo) #! 00:00
    else:
        return "00:0" + str(segundo) #! 00:00`
        
Essas alterações estão testadas e funcionais com mapa.py, conforme conversamos antes.

AgathaLice replied:
![tempo](https://github.com/user-attachments/assets/8d15367a-5476-47e8-9474-859689fed6cb)

Está assim agora, mais algo?

AgathaLice replied:
Consertei algo que não havia percebido, aparecia 2:34 por exemplo ao invés de 02:34, agora está correto.

AgathaLice committed:
update timer.py

Correção  do retorno da aparência do tempo.
Agora é 00:00

AgathaLice committed:
Update timer.py

Tempo de jogo agora são 90 segundos (um minuto e meio) e retirada a linha de teste
